docs/azure.rst

```diff
@@ -275,6 +275,9 @@ azure.batch.allowPoolCreation                   Enable the automatic creation of
 azure.batch.deleteJobsOnCompletion              Enable the automatic deletion of jobs created by the pipeline execution (default: ``true``).
 azure.batch.deletePoolsOnCompletion             Enable the automatic deletion of compute node pools upon pipeline completion (default: ``false``).
 azure.batch.copyToolInstallMode                 Specify where the `azcopy` tool used by Nextflow. When ``node`` is specified it's copied once during the pool creation. When ``task`` is provider, it's installed for each task execution (default: ``node``).
+azure.batch.pools.<name>.publisher              Specify the publisher of virtual machine type used by the pool identified with ``<name>`` (default ``microsoft-azure-batch``).
+azure.batch.pools.<name>.offer                  Specify the offer type of the virtual machine type used by the pool identified with ``<name>`` (default ``centos-container``).
+azure.batch.pools.<name>.sku                    Specify the ID of the Compute Node agent SKU which the pool identified with ``<name>`` supports (default ``batch.node.centos 8``).
```

Comment:
Do you mean to document these settings?

If so, the question should be: are we going to always use centOS node or do we want to let the user also select if they want to run on an Ubuntu or Debian node? If we are good with centOS, these should be removed from the doc. Rather, my assumption was that centOS is the default, but the user can decide to change (since Azure Batch lets you do so) OS and version.

This also affects the PR on Azure Files: if we use only centOS, then the mount path is fixed, no need to add it to the pool settings (sorry to mix up the PRs, but I feel the decision to take here needs a bigger picture).

Reply:
Oops I meant to highlight only the `sku`

Reply:
If we don't expose the SKU, how the user can choose (for instance) between these 2 image references?

```
sku: batch.node.centos 7
osType: linux
offer: centos-container
publisher: microsoft-azure-batch
```

```
sku: batch.node.centos 8
osType: linux
offer: centos-container
publisher: microsoft-azure-batch
```

Also, remember, it's optional, like the rest of the image reference information in the config file. I do expect most users won't change the defaults.

Reply:
Fair enough

Reply:
is there any Azure docs that could be linked as reference for the above option? otherwise it's obscure from where these values come from

Reply:
Unfortunately, the supported images change from region to region and over time.

The Azure doc [here](https://docs.microsoft.com/en-us/azure/batch/batch-linux-nodes#list-of-virtual-machine-images) only mention to go in the Azure Marketplace or call `listImageReference()` on the `BatchClient` (like NF does) to fetch the current list of supported images in your region.

Would adding a small section in the Azure page with the 3 most common configs (Ubuntu, CentOS and Debian) simplify the user's life here?

Reply:
Yeah, providing some concrete alternative can be useful both for `offer` and `sku` (btw what does acronym  mean?)

Reply:
Ok, I'll add to the doc these common alternatives.

SKU should stand for "Stock Keeping Unit".

Reply:
I added the section "Pool nodes". See what you think.